packages/expo-media-library/ios/EXMediaLibrary/EXMediaLibrary.m

```diff
@@ -554,6 +570,19 @@ - (void)photoLibraryDidChange:(PHChange *)changeInstance
         }
         
         [_eventEmitter sendEventWithName:EXMediaLibraryDidChangeEvent body:body];
+        return;
+      }
+      
+      if (@available(ios 14, *)) {
+        // Emit event when the user changed the limited permissions.
+        if (!changeDetails.hasIncrementalChanges) {
+          [_eventEmitter sendEventWithName:EXMediaLibraryDidChangeEvent body:@{
+            @"shouldFullyReload": @(true),
+            @"insertedAssets": @[],
+            @"deletedAssets": @[],
+            @"updatedAssets": @[]
```

Comment:
I know this might be confusing, but what about splitting this into the union type:
```
| {
    insertedAssets: Assets[],
    deletedAssets: Assets[],
    updatedAssets: Assets[]
  }
| {
    shouldFullyReload: boolean // this name should change
  }
 
```

Reply:
I'm not a big fan of this. However, I split it to:
```
  /**
   * iOS only
   */
  | {
      assetPermissionsChanged: boolean;
      insertedAssets: Asset[];
      deletedAssets: Asset[];
      updatedAssets: Asset[];
    }
  | object;
```